### PR TITLE
Add pit for multi query

### PR DESCRIPTION
**Pull request by rupal-bq**
### Description
- Added option to use search after with pit instead of scroll api for multi query
 
### Issues Resolved
https://github.com/opensearch-project/sql/issues/2603
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: legacy/src/main/java/org/opensearch/sql/legacy/executor/multi/MinusExecutor.java [hunk 1]
```diff
@@ -63,45 +72,63 @@ public MinusExecutor(Client client, MultiQueryRequestBuilder builder) {
...
+        pit =
+            new PointInTimeHandlerImpl(
+                client,
+                ArrayUtils.concat(
+                    builder.getOriginalSelect(true).getIndexArr(),
+                    builder.getOriginalSelect(false).getIndexArr()));
+        pit.create();
+      }
+
+      if (this.useTermsOptimization && this.fieldsOrderFirstTable.length != 1) {
```
Comment (dai-chen) [inline legacy/src/main/java/org/opensearch/sql/legacy/executor/multi/MinusExecutor.java:86]: The logic starting from this become even more complex than before. Could you explain why and think about if any better approach?

File: legacy/src/main/java/org/opensearch/sql/legacy/executor/multi/MinusExecutor.java [hunk 2]
```diff
@@ -49,6 +57,7 @@ public class MinusExecutor implements ElasticHitsExecutor {
   private String[] fieldsOrderFirstTable;
   private String[] fieldsOrderSecondTable;
   private String seperator;
+  private PointInTimeHandler pit;
```
Comment (dai-chen) [inline legacy/src/main/java/org/opensearch/sql/legacy/executor/multi/MinusExecutor.java:60]: If all executor depends on PIT, `getResponseWithHits` and others like `Client`, probably we can move all common fields to `ElasticHitsExecutor` and make it abstract class?

File: legacy/src/main/java/org/opensearch/sql/legacy/executor/multi/MinusExecutor.java (hunk 1 above)
Comment (dai-chen) [reply legacy/src/main/java/org/opensearch/sql/legacy/executor/multi/MinusExecutor.java:79]: I see. I recall we have this similar logic in your previous PR? If all `ElasticHitsExecutor` has such logic, can we move it to parent class?

File: legacy/src/main/java/org/opensearch/sql/legacy/executor/ElasticHitsExecutor.java [hunk 3]
```diff
@@ -5,13 +5,77 @@
...
+import org.opensearch.search.builder.PointInTimeBuilder;
+import org.opensearch.sql.legacy.domain.Select;
+import org.opensearch.sql.legacy.esdomain.LocalClusterState;
 import org.opensearch.sql.legacy.exception.SqlParseException;
+import org.opensearch.sql.legacy.pit.PointInTimeHandler;
 
 /** Created by Eliran on 21/8/2016. */
-public interface ElasticHitsExecutor {
-  void run() throws IOException, SqlParseException;
+public abstract class ElasticHitsExecutor {
```
Comment (dai-chen) [inline legacy/src/main/java/org/opensearch/sql/legacy/executor/ElasticHitsExecutor.java:28]: np: could you add missing Javadoc?

File: legacy/src/main/java/org/opensearch/sql/legacy/executor/multi/MinusExecutor.java (hunk 1 above)
Comment (rupal-bq) [reply legacy/src/main/java/org/opensearch/sql/legacy/executor/multi/MinusExecutor.java:79]: No, not all ElasticHitsExecutor has similar logic. Join and multiquery has this pit logic but there is no scrolling in UnionExecutor so it doesn't have such logic.

**Review by dai-chen — APPROVED** [review]
Approved. This https://github.com/opensearch-project/sql/pull/2753/files#r1669287520 is still open.

**Review by vamsi-amazon — APPROVED** [review]

File: legacy/src/main/java/org/opensearch/sql/legacy/executor/ElasticHitsExecutor.java (hunk 3 above)
Comment (rupal-bq) [reply legacy/src/main/java/org/opensearch/sql/legacy/executor/ElasticHitsExecutor.java:28]: done

**Pull request merged**